src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/KpisInfoProcessor.java

```diff
@@ -0,0 +1,46 @@
+package cz.centrum.haffner.SimpleTrainingDavid.AppServices;
+
+import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.KpisInfoData;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+public class KpisInfoProcessor {
+    private Set differentOriginsSet = new HashSet<Integer>();
```

Comment:
Takhle si zadefinuješ Raw type použití.
Správně bys měl typ kolekce uvést vlevo:
private Set<Integer> differentOriginsSet = new HashSet<>();

Reply:
předěláno ...